mint/config.go

```
@@ -54,3 +65,177 @@ type MintLimits struct {
 	MintingSettings MintMethodSettings
 	MeltingSettings MeltMethodSettings
 }
+
+func ConfigFromEnv() (*Config, error) {
+	var inputFeePpk uint = 0
+	if inputFeeEnv, ok := os.LookupEnv("INPUT_FEE_PPK"); ok {
+		fee, err := strconv.ParseUint(inputFeeEnv, 10, 16)
+		if err != nil {
+			return nil, fmt.Errorf("invalid INPUT_FEE_PPK: %w", err)
+		}
+		inputFeePpk = uint(fee)
+	}
+
+	derivationPathIdx, err := strconv.ParseUint(os.Getenv("DERIVATION_PATH_IDX"), 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("invalid DERIVATION_PATH_IDX: %w", err)
+	}
+
+	port := os.Getenv("MINT_PORT")
+	if len(port) == 0 {
+		port = "3338"
+	}
+
+	mintPath := os.Getenv("MINT_DB_PATH")
+	// if MINT_DB_PATH is empty, use $HOME/.gonuts/mint
+	if len(mintPath) == 0 {
+		homedir, err := os.UserHomeDir()
+		if err != nil {
+			return nil, err
+		}
+		mintPath = filepath.Join(homedir, ".gonuts", "mint")
+	}
```

if moving this to the mint package, I don't think it should default to this directory. Think it would be better to return an error if empty and then application-specific can provide a default directory if the error happens.